Add support for multiple Robinhood accounts

superpowered opened the pull request:
Adds support for multiple Robinhood accounts. (closes #16)

- [ ] Should be able to Enable / Disable Multiple Accounts.
- [ ] Should be able to Add rows of accounts.
- [ ] Should be able to delete rows of accounts. (Excluding the first row)
- [ ] Data should end up stored as an array of objects `[{ robinHoodValue: '' }]`
- [ ] Data should be sanitized before output
- [ ] Validations should fire on fields if they are blank. (more validations could probably be added here)
- [ ] Should be prompted to set up after toggling Multiple Accounts on or off
- [ ] After Setup of multiple, should be told to remove old single properties
- [ ] After Setup of single, should be told to remove old multiple properties
- [ ] Should only update 4 correct properties on update
- [ ] Should prompt for setup if one or more properties expected is missing
- [ ] Should throw up a warning if a property was found, but could not be updated with the data provided.

Other tidbits:
- Moves `urls` & `interfaces` to `constants` folder
- Removes .`vscode/settings.json`
- Adds .`vscode/settings.json.default` so devs can opt in to the projects default settings but still overwrite if needed.
- Adds .`vscode/extensions.json` recomending eslint as an extension.
- `lint:fix` command added for quick linting.

superpowered committed:
Add Multiple Accounts to Settings page. Should allow to add and delete accounts.
Data should end up stored as an array of objects `[{ robinHoodValue: '', mintValue: '' }]`
Currently does NOT sanitize data
Currently does NOT validate input

superpowered committed:
Linting

superpowered committed:
re-indent head/body. formatOnSave causing issues here.

superpowered committed:
Stop multiple event handlers from being registered on add button

Comments.

pkmnct commented:
Just a heads-up, I was thinking about scraping the Robinhood account username and using it as a name on the properties, as a way of allowing multiple accounts to be set up.

superpowered committed:
Get account name from robin hood account screen.
Some basic sanitization.
Error color.
Re adding `mint-force-sync`, may need to be removed on merge

superpowered committed:
lint

superpowered committed:
Update fields based on account name if we find a matching one.

superpowered committed:
Can pass updated account name to success message. Definately needs cleaning up and sanitization.

superpowered committed:
Merge commit 'c8ab2b380d1330a3ddac6caaa7a78b1487ae6660' into multiple-accounts

# Conflicts:
#	public/html/settings.html

superpowered committed:
Mint value is unneeded

superpowered committed:
Add Check / create for multiple accounts.

superpowered commented:
Okay. Definitely will need a bit of cleanup, but should be functional at this point.

superpowered committed:
Stop. Un.in.dent.ing.

superpowered committed:
.vscode to gitignore. Should keep the original file, but allow it to be overridden. (Turn off things like formatOnSave)

superpowered committed:
rm .vscode/settings.json

superpowered committed:
Merge commit '1a10de5d0bbc6e6602076cae0bfcebe87e3cd906' into multiple-accounts

# Conflicts:
#	.gitignore
#	.vscode/settings.json
#	public/html/settings.html

superpowered committed:
Pretty sure stellar wasn't supposed to get commited.

pkmnct commented on .gitignore:
```diff
@@ -10,4 +10,7 @@ node_modules
 
-.stellar
\ No newline at end of file
+.vscode/*
```
This directory contains the VS Code workspace settings, and is intended to be in the repository.

superpowered replied:
The problem with committing the `.vscode` folder is it means none of those settings can be overwritten locally since they are "stronger" than user level settings. 
This can cause issues when settings differ from user to user. (example: Your vscode eslint seems to function different than mine locally, and combined with "formatOnSave" causing my linter to unindent the .json and html files every time I touch them) 
<img width="433" alt="Screen Shot 2021-01-19 at 5 56 40 AM" src="https://user-images.githubusercontent.com/7035467/105032325-3fbf0700-5a1c-11eb-806f-fdccc243d83d.png">

I've put back in `.vscode/settings.json.default` So the settings can be inherited if needed. The eslint stuff really doesn't need to be forced on each user on save, since the pipeline should be configured to catch lint errors, and the actual formatting can be handled via the command line :fix actions.

pkmnct replied:
I'll have to put some thought into this. I do realize that I did not include the "recommended extensions," which allow these settings to work as intended. I think that I'd prefer, if you are using VS Code already, that files be saved the exact same way without a manual step that could be forgotten (besides the initial extension setup).

pkmnct replied:
I like the workaround

superpowered committed:
Allow default

pkmnct commented:
I'm going to plan to take a look a this over the weekend!

superpowered commented:
> I'm going to plan to take at look a this over the weekend!

Yeah, hopefully I'll get to look at it some more and clean it up before then. A bit more swamped than usual due to the shorter work week.

superpowered committed:
Recommend eslint to match recommended settings

superpowered committed:
Merge commit '4ea984ab381515819a83142426750e174cb95ea6' into multiple-accounts

# Conflicts:
#	public/html/settings.html
#	src/background/main.ts
#	src/content/mint/properties/check.ts
#	src/content/mint/properties/update.ts

superpowered committed:
Clean up the update script

superpowered committed:
Move Interfaces & urls to constants

superpowered committed:
Add lint:fix, run it

superpowered committed:
Commenting.

superpowered committed:
rename match to isMultipleAccounts. reorder functions by priority

superpowered committed:
Browser typing is a dev extension

superpowered committed:
Multiple Accounts readme

superpowered committed:
can remove `mint-force-sync`

superpowered committed:
Move `sanitizeInput` to utilities, `robinHoodValue` renamed to `robinHoodAccountName`

superpowered committed:
Clean up settings

superpowered committed:
Clean up check

superpowered committed:
Some final cleanup

superpowered committed:
Error with missing args on timeout

superpowered committed:
Fix issue with other not being set

superpowered committed:
Warnings for if a property cant be found or set

pkmnct committed:
Add prettier to recommended extensions

pkmnct committed:
Run `npm i` to ensure package-lock.json up-to-date

pkmnct committed:
Style/formatting changes, other minor tweaks

pkmnct commented on public/html/settings.html:
```diff
@@ -69,6 +78,38 @@ <h2>Trigger Initial Setup</h2>
+              <div class="mutiple-accounts-input-wrapper">
+                <label>
+                  RobinHood Account Name
+                  <input required type="text" class="setting-multipleAccounts-robinHoodInput" />
```
Rather than having the user manually enter their account names, we can just scrape the account name every time we sync Robinhood. If the multiple account setting is not enabled, the value is ignored.

If it is enabled, we do a setup check (similar to the initial setup), where we check if this device has synced with the Robinhood account logged in. (These could be stored in an array in chrome.storage.sync). If the current device has not set up that account before, we can then trigger the property setup/check.

This would give users a more seamless experience, and reduce logic on the settings page since it would just be a checkbox.

superpowered replied:
Alright, latest push should accomplish this... 
Need to probably test some of the flows and make sure the notifications seem correct in each one.

... Probably also need to add a delete script  under the debug at some point, more just for convienence, since adding and deleting all these accounts is crazy tedious.

Changing the `returnValue.accountName = accountNameElement.innerText;` to a static value works well enough for testing to add new accounts.

superpowered committed:
Remove manual adding of accounts, now sync script will check if account is new and add automatically.

superpowered committed:
Some debugs for the new flows

pkmnct committed:
Minor cleanup

pkmnct committed:
Fix typo

pkmnct committed:
message may not have event key

pkmnct committed:
Remove callbackData

pkmnct committed:
Bump version

pkmnct committed:
Fix waitForElement

pkmnct committed:
Add additional default settings

pkmnct commented:
Closes #16

superpowered committed:
Merge commit '35de422d2a604bcbf7dafa2788f8521f398357a6' into multiple-accounts

# Conflicts:
#	package-lock.json
#	package.json
#	public/manifest.json
#	src/background/main.ts
#	src/content/mint/properties/update.ts
#	src/content/robinhood/main.ts
